src/api/goods.ts

```
@@ -4,13 +4,37 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      }
+
+      return 'A API está inacessível via protocolo HTTP.';
+    })
+    .catch(error => error);
 }
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+export const get5First = (): Promise<Good[]> => {
+  return getAll()
+    .then((goods: Good[]): Good[] => {
+      const goodsWrapper = [...goods];
+
+      goodsWrapper.sort((good1: Good, good2: Good) => {
+        return good1.name.localeCompare(good2.name);
+      });
+
+      return goodsWrapper.slice(0, 5);
+    })
+    .catch(error => error);
 };
 
-export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+export const getRedGoods = (): Promise<Good[]> => {
+  return getAll()
+    .then((goods: Good[]): Good[] => {
+      return goods.filter((good: Good) => {
+        return good.color === 'red';
+      });
+    })
+    .catch(error => error);
```

The .catch block returns the error instead of rejecting or throwing. This means the promise resolves to an error object instead of rejecting, which is not standard and makes error handling in the UI unreliable. You should throw the error or let the promise reject. This violates checklist item #3: "Error handling must be implemented for server requests (e.g., handle network or response errors and reflect them appropriately in the app logic/UI)."